sonar-plugin-api/src/main/java/org/sonar/api/ce/posttask/PostProjectAnalysisTaskTester.java

```diff
@@ -0,0 +1,530 @@
+ *           .setUuid("uuid")
+ *           .setKey("key")
+ *           .setName("name")
+ *           .build())
+ *       .at(new Date())
+ *       .withQualityGate(
+ *         newQualityGateBuilder()
+ *           .setId("id")
+ *           .setName("name")
+ *           .setStatus(QualityGate.Status.OK)
+ *           .add(
+ *             newConditionBuilder()
+ *               .setMetricKey("metric key")
+ *               .setOperator(QualityGate.Operator.GREATER_THAN)
+ *               .setErrorThreshold("12")
+ *               .setOnLeakPeriod(true)
+ *               .build(QualityGate.EvaluationStatus.OK, "value"))
+ *           .build())
+ *       .execute();
+ *
+ *     assertThat(postProjectAnalysisTask.projectAnalysis).isNotNull();
+ *   }
+ * }
+ * }
+ * </pre>
+ */
+@Beta
+public class PostProjectAnalysisTaskTester {
+  private static final String DATE_CAN_NOT_BE_NULL = "date can not be null";
+  private static final String PROJECT_CAN_NOT_BE_NULL = "project can not be null";
+  private static final String CE_TASK_CAN_NOT_BE_NULL = "ceTask can not be null";
+
+  private final PostProjectAnalysisTask underTest;
+  @CheckForNull
+  private CeTask ceTask;
+  @CheckForNull
+  private Project project;
+  @CheckForNull
+  private Date date;
+  @CheckForNull
+  private QualityGate qualityGate;
+
+  private PostProjectAnalysisTaskTester(PostProjectAnalysisTask underTest) {
+    this.underTest = requireNonNull(underTest, "PostProjectAnalysisTask instance can not be null");
+  }
+
+  public static PostProjectAnalysisTaskTester of(PostProjectAnalysisTask underTest) {
+    return new PostProjectAnalysisTaskTester(underTest);
+  }
+
+  public static CeTaskBuilder newCeTaskBuilder() {
+    return new CeTaskBuilder();
+  }
+
+  public static ProjectBuilder newProjectBuilder() {
+    return new ProjectBuilder();
+  }
+
+  public static QualityGateBuilder newQualityGateBuilder() {
+    return new QualityGateBuilder();
+  }
+
+  public static ConditionBuilder newConditionBuilder() {
+    return new ConditionBuilder();
+  }
+
+  public PostProjectAnalysisTaskTester withCeTask(CeTask ceTask) {
+    this.ceTask = requireNonNull(ceTask, CE_TASK_CAN_NOT_BE_NULL);
+    return this;
+  }
+
+  public PostProjectAnalysisTaskTester withProject(Project project) {
+    this.project = requireNonNull(project, PROJECT_CAN_NOT_BE_NULL);
+    return this;
+  }
+
+  public PostProjectAnalysisTaskTester at(Date date) {
+    this.date = requireNonNull(date, DATE_CAN_NOT_BE_NULL);
+    return this;
+  }
+
+  public PostProjectAnalysisTaskTester withQualityGate(@Nullable QualityGate qualityGate) {
+    this.qualityGate = qualityGate;
+    return this;
+  }
+
+  public void execute() {
+    this.ceTask = requireNonNull(ceTask, CE_TASK_CAN_NOT_BE_NULL);
+    this.project = requireNonNull(project, PROJECT_CAN_NOT_BE_NULL);
+    this.date = requireNonNull(date, DATE_CAN_NOT_BE_NULL);
+
+    this.underTest.finished(
+      new PostProjectAnalysisTask.ProjectAnalysis() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this anonymous class number of lines from 31 to at most 20, or make it a named class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1188)